### PR TITLE
HubConnection implements IAsyncDisposable

### LINKED ISSUE #13082: Best practice to dispose SignalR HubConnection in .NET Core
Is there a best practice or suggestion for how to stop/dispose `HubConnection`?

Since `HubConnection` is not `IDisposable`, we cannot use `using`, any suggestion if we need to use `StopAsync` and `DisposeAsync` explicitly?

Any performance impact to use or not to use `StopAsync` and `DisposeAsync`?

From the [doc](https://docs.microsoft.com/en-us/aspnet/core/signalr/dotnet-client?view=aspnetcore-2.2), there is no mention of whether we need or should dispose the `HubConnection`

### PR DESCRIPTION
**Pull request by BrennanConroy**
Fixes https://github.com/aspnet/AspNetCore/issues/13082

Since we're doing this in 5.0 we can make the break?!

### CODE REVIEW

**Review by halter73 — APPROVED** [review]

File: src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs [hunk 1]
```diff
@@ -292,8 +291,8 @@ public async Task StopAsync(CancellationToken cancellationToken = default)
         /// <summary>
         /// Disposes the <see cref="HubConnection"/>.
         /// </summary>
-        /// <returns>A <see cref="Task"/> that represents the asynchronous dispose.</returns>
-        public async Task DisposeAsync()
+        /// <returns>A <see cref="ValueTask"/> that represents the asynchronous dispose.</returns>
+        public async ValueTask DisposeAsync()
```
Comment (JamesNK) [inline src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: You *could* do this without a breaking change by making a new explicit method:

```csharp
async ValueTask IAsyncDisposable.DisposeAsync()
{
    // Do the things
}

public Task DisposeAsync()
{
    return ((IAsyncDisposable)this).DisposeAsync().AsTask();
}
```

Comment (BrennanConroy) [reply src/SignalR/clients/csharp/Client.Core/src/HubConnection.cs]: Yeah, I did that in 3.0 but we decided to move it to 5.0 and we can make the small break here.

**Comment by BrennanConroy** [comment]
@aspnet-hello triage https://dev.azure.com/dnceng/public/_build/results?buildId=312261

**Comment by aspnet-hello** [comment]
This comment was made automatically. If there is a problem contact aspnetcore-build@microsoft.com.

I've triaged the above build. I've created/commented on the following issue(s)
https://github.com/aspnet/AspNetCore-Internal/issues/2674
https://github.com/aspnet/AspNetCore-Internal/issues/2617
https://github.com/aspnet/AspNetCore-Internal/issues/1323

**Pull request merged**